### PR TITLE
Misc/trace directions debug symbols

### PR DESCRIPTION
**Pull request by rrahn**
Adds debug symbols for printing trace matrix with combinations of horizontal and vertical gap open.

Only last commit is relevant.

~~Blocked by #1229~~

### CODE REVIEW

File: test/unit/alignment/matrix/debug_stream_debug_matrix_test.cpp [hunk 1]
```diff
@@ -135,75 +135,75 @@ struct debug_matrix_stream_test : public ::testing::Test
...
+        " ║⇡  ║↖⇡ ║↖  ║↖⇡⇠║↖  ║↖⇡⇠║↖  ║⇡  ║↖  ║↖  ║↖⇠ ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║⇠  ║\n"
         " ╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬\n"
-        " ║↑  ║↑  ║↑  ║↖  ║↑← ║↖  ║←  ║↖↑←║↖↑ ║↖↑ ║↖  ║↖  ║↖← ║←  ║←  ║←  ║←  ║\n"
+        " ║⇡  ║⇡  ║⇡  ║↖  ║⇡⇠ ║↖  ║⇠  ║↖⇡⇠║↖⇡ ║↖⇡ ║↖  ║↖  ║↖⇠ ║⇠  ║⇠  ║⇠  ║⇠  ║\n"
         " ╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬\n"
-        " ║↑  ║↑  ║↑  ║↑  ║↖  ║↑  ║↖  ║←  ║←  ║↖↑←║↖↑ ║↖↑ ║↖  ║↖  ║↖← ║←  ║←  ║\n"
+        " ║⇡  ║⇡  ║⇡  ║⇡  ║↖  ║⇡  ║↖  ║⇠  ║⇠  ║↖⇡⇠║↖⇡ ║↖⇡ ║↖  ║↖  ║↖⇠ ║⇠  ║⇠  ║\n"
         " ╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬═══╬\n"
-        " ║↺  ║↑  ║↑  ║↑  ║↖↑ ║↖↑ ║↑  ║↖  ║↖← ║←  ║←  ║↖↑←║↖↑ ║↖↑ ║↖  ║↖  ║↖← ║\n"
+        " ║↺  ║⇡  ║⇡  ║⇡  ║↖⇡ ║↖⇡ ║⇡  ║↖  ║↖⇠ ║⇠  ║⇠  ║↖⇡⇠║↖⇡ ║↖⇡ ║↖  ║↖  ║↖⇠ ║\n"
```
Comment (h-2) [inline test/unit/alignment/matrix/debug_stream_debug_matrix_test.cpp]: Why these changes?

Comment (rrahn) [reply test/unit/alignment/matrix/debug_stream_debug_matrix_test.cpp]: We had in the first implementation only considered normal gaps and now added versions for gap_open and gap_extension. Since the test only uses gap_extensions it is not visible at this places. But we decided to have the solid arrow for gap_opening and the dotted arrow for gap extension as it seems more natural for the association with gap opening and extension.

File: include/seqan3/alignment/pairwise/alignment_result.hpp [hunk 2]
```diff
@@ -190,7 +199,7 @@ class alignment_result
 
     /*!\brief Returns the actual alignment, i.e. the base pair matching.
      * \return At least two aligned sequences, which represent the alignment.
-     * \attention This function with fail the compilation, if the alignment was not requested in the alignment
+     * \attention Compiling this function will fail, if the alignment was not requested in the alignment
```
[the remark this answers is not in the document]
Comment (rrahn) [reply include/seqan3/alignment/pairwise/alignment_result.hpp]: Ok, the description might not be so easy to understand for users. In general it follows a policy based design. If the policy for computing an alignment was requested this function will be available. If not the function is not available and the compiler will notify this with a static assert. But you merged this change already in another PR. I will later adapt the writing of this.

File: include/seqan3/core/type_list/traits.hpp [hunk 3]
```diff
@@ -329,6 +329,22 @@ auto split_after(type_list<pack1_t...>, type_list<head_t, pack2_t...>)
...
+ * \ingroup type_list
+ */
+template <typename replace_t,
+          typename ...pack1_t,
+          typename head_t,
+          typename ...pack2_t>
+auto replace_at(std::pair<type_list<pack1_t...>, type_list<head_t, pack2_t...>>)
+{
+    return type_list<pack1_t..., replace_t, pack2_t...>{};
+}
```
[the remark this answers is not in the document]
Comment (rrahn) [reply include/seqan3/core/type_list/traits.hpp]: Ok, I agree. BUt it was already merged in another PR. I will see that I make a little fix for this.

File: include/seqan3/alignment/matrix/trace_directions.hpp [hunk 4]
```diff
@@ -56,17 +56,37 @@ namespace seqan3
  * \param s The seqan3::debug_stream.
  * \param trace The trace direction.
  * \relates seqan3::debug_stream_type
+ *
+ * \details
+ *
+ * The following table shows the printed symbol of a particular seqan3::detail::trace_directions:
```
Comment (h-2) [inline include/seqan3/alignment/matrix/trace_directions.hpp]: This sentence highlights why we use generic singular. The enum should be called `seqan3::detail::trace_direction` but thats a different issue.

File: include/seqan3/alignment/matrix/trace_directions.hpp [hunk 5]
```diff
@@ -56,17 +56,37 @@ namespace seqan3
...
+    static char const * csv[32]{ "N",   "D",   "U",   "DU",   "u",   "Du",   "Uu",   "DUu",
+                                 "L",  "DL",  "UL",  "DUL",  "uL",  "DuL",  "UuL",  "DUuL",
+                                 "l",  "Dl",  "Ul",  "DUl",  "ul",  "Dul",  "Uul",  "DUul",
+                                "Ll", "DLl", "ULl", "DULl", "uLl", "DuLl", "UuLl", "DUuLl"};
 
     bool is_unicode = (s.flags2() & fmtflags2::utf8) == fmtflags2::utf8;
     auto const & trace_dir = is_unicode ? unicode : csv;
 
-    s << trace_dir[static_cast<size_t>(trace) % 8u];
+    s << trace_dir[static_cast<size_t>(trace) % 32u];
```
Comment (h-2) [inline include/seqan3/alignment/matrix/trace_directions.hpp]: why the modulo

Comment (rrahn) [reply include/seqan3/alignment/matrix/trace_directions.hpp]: well it is not necessary anymore. good point.

File: include/seqan3/alignment/matrix/trace_directions.hpp (hunk 4 above)
Comment (rrahn) [reply include/seqan3/alignment/matrix/trace_directions.hpp]: I know, something that happens sometimes later together with a bunch of clean up steps
